coordinator/src/test/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlannerSpec.scala

```diff
@@ -109,9 +111,7 @@ class ShardKeyRegexPlannerSpec extends FunSpec with Matchers with ScalaFutures {
     val lp = Parser.metadataQueryToLogicalPlan("http_requests_total{job=\"prometheus\", method=\"GET\"}",
       TimeStepParams(1000, 1000, 3000))
 
-    val shardKeyMatcher = (shardColumnFilters: Seq[ColumnFilter]) => { Seq(Seq(ColumnFilter("_ws_", Equals("demo")),
-      ColumnFilter("_ns_", Equals("App-1"))), Seq(ColumnFilter("_ws_", Equals("demo")),
-      ColumnFilter("_ns_", Equals("App-2"))))}
+    val shardKeyMatcher = (shardColumnFilters: Seq[ColumnFilter], query: String) => { Seq.empty }
```

Comment:
Even though query does not have regex, shardKeyMatcher should not be empty. It should just return shard key filters which are present since logicalPlan filters are replaced with  output of shardKeyMatcher
`shardKeyMatcher(nonMetricShardKeyFilters).map(f =>
                           queryPlanner.materialize(logicalPlan.replaceFilters(f), qContext))`

Reply:
Correct there will be atleast one `ShardKeyMatcherResult` when proper `nonMetricShardKeyFilters` are passed.